contracts/Marketplace.sol

```diff
@@ -87,18 +107,20 @@ contract Marketplace is Proofs, StateRetrieval, Endian {
   }
 
   function requestStorage(Request calldata request) public {
-    require(request.client == msg.sender, "Invalid client address");
+    if (request.client != msg.sender) {
+      revert Marketplace_InvalidClientAddress();
+    }
 
     RequestId id = request.id();
-    require(_requests[id].client == address(0), "Request already exists");
-    require(
-      request.expiry > 0 && request.expiry < request.ask.duration,
-      "Expiry not in range"
-    );
-    require(
-      request.ask.maxSlotLoss <= request.ask.slots,
-      "maxSlotLoss exceeds slots"
-    );
+    if (_requests[id].client != address(0)) {
+      revert Marketplace_RequestAlreadyExists();
+    }
+    if (request.expiry == 0 || request.expiry > request.ask.duration) {
```

Comment:
There is a subtle change here, where request.expiry == request.ask.duration was previously not allowed, but it is allowed now. Not a problem, but something to keep in mind when integrating this into nim-codex. Maybe there's an integration test that depends on the previous behaviour.

Reply:
It's probably better to change it back, imo.

Reply:
Very good catch, I'm gonna change this.